plugins/net_plugin/net_plugin.cpp

```diff
@@ -2203,6 +2208,18 @@ namespace eosio {
               sync_next_expected_num < sync_last_requested_num );
    }
 
+   bool sync_manager::is_sync_request_ahead_allowed(block_num_type blk_num) const REQUIRES(sync_mtx) {
+      if (blk_num >= sync_last_requested_num) {
+         // do not allow to get too far ahead (sync_fetch_span) of chain head
+         auto fetch_span = active_sync_fetch_span(false);
```

Comment:
I think this is still not right because it takes into account `conf.max_reversible_blocks`

Reply:
See https://github.com/AntelopeIO/spring/pull/652